### PR TITLE
Tests for View Command

### PR DESCRIPTION
**Pull request by russelltankaimin**
## Testing for View Command
- [X] Tests for View Command
- [x] Tests for ViewCommandParser 

### This PR is ready for review!

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 1]
```diff
@@ -64,4 +64,26 @@ public Optional<Person> retrievePerson(Model model) {
...
+        if (other == this) {
+            return true;
+        }
+        if (other instanceof ViewCommand) {
+            ViewCommand otherCommand = (ViewCommand) other;
+            return otherCommand.getIndex().equals(getIndex())
+                    && otherCommand.getName().equals(getName());
+        } else {
+            return false;
+        }
```
Comment (mynameizzhafeez) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:87]: Good use of happy path.

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        ViewCommand command = new ViewCommand(null, null);
+        CommandResult result = command.execute(model);
+        assertEquals(LINUS, result.getDisplayPerson());
+    }
+
+    @Test
+    public void execute_noSuchName_userProfileDisplayed() throws CommandException {
+        ViewCommand command1 = new ViewCommand("Lagrange", null);
+        ViewCommand command2 = new ViewCommand("Alan Turing", null);
+        ViewCommand command3 = new ViewCommand("Pierre de Fermat", null);
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:70]: Could possibly catch for partial names like "Edward" but this should be enough already. Next time if we doing comprehensive testing can do all at once.

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+        CommandResult result1 = command1.execute(model);
+        CommandResult result2 = command2.execute(model);
+        CommandResult result3 = command3.execute(model);
+
+        assertEquals(LINUS, result1.getDisplayPerson());
+        assertEquals(LINUS, result2.getDisplayPerson());
+        assertEquals(LINUS, result3.getDisplayPerson());
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:78]: LGTM. I understand what the command does already.

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    public void compare_viewCommandInequality_false() {
+        ViewCommand command1 = new ViewCommand("Lisa Meitner", null);
+        ViewCommand command2 = new ViewCommand("Lisa Meitner", Index.fromZeroBased(1));
+        assertNotEquals(command1, command2);
+
+        ViewCommand command3 = new ViewCommand(null, Index.fromZeroBased(1));
+        ViewCommand command4 = new ViewCommand(null, null);
+        assertNotEquals(command3, command1);
+        assertNotEquals(command3, command2);
+        assertNotEquals(command3, command4);
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:115]: Good coverage of edge cases!

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+        ViewCommand command5 = new ViewCommand("Marie Curie", null);
+        ViewCommand command6 = new ViewCommand(null, Index.fromZeroBased(2));
+        assertNotEquals(command5, command1);
+        assertNotEquals(command5, command1);
+        assertNotEquals(command6, command2);
+        assertNotEquals(command6, command1);
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:122]: Here as well~

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    @Test
+    public void compare_otherCommandInequality_false() {
+        ViewCommand viewCommand = new ViewCommand("Lisa Meitner", null);
+        AddCommand addCommand = new AddCommand(ALBERT);
+        DeleteCommand deleteCommand = new DeleteCommand(Index.fromZeroBased(1));
+        assertNotEquals(addCommand, viewCommand);
+        assertNotEquals(deleteCommand, viewCommand);
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:131]: Hmm interesting catch, but LGTM.

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+    }
+
+    @Test
+    public void compare_foreignObjectInequality_false() {
+        ViewCommand viewCommand = new ViewCommand("Lisa Meitner", null);
+        assertNotEquals("Hello", viewCommand);
+    }
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:139]: Intellij is probably smart enough to label this as a redundant test, but LGTM.

File: src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test
+    public void parse_indexOnly_success() {
+        for (int i = 1; i < 100; i++) {
+            Index index = Index.fromOneBased(i);
+            assertParseSuccess(viewCommandParser,
+                    String.valueOf(i),
+                    new ViewCommand(null, index));
+        }
```
Comment (mynameizzhafeez) [inline src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java:67]: Parse index can have more opportunity for failures, so in the future can check for invalid index.

**Review by mynameizzhafeez — APPROVED** [review]
LGTM.

File: src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java (hunk 8 above)
Comment (russelltankaimin) [reply src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java:67]: @mynameizzhafeez I have tried that. However, ```Index``` would have caught all the bad cases already. We would essentially be repeating test cases as such. This is an advantage in using ```Index``` over ```int``` for this case.
